src/wp-includes/post.php

```diff
@@ -7893,6 +7893,27 @@ function _prime_post_caches( $ids, $update_term_cache = true, $update_meta_cache
 
 		update_post_caches( $fresh_posts, 'any', $update_term_cache, $update_meta_cache );
 	}
+
+	if ( ! $update_meta_cache && ! $update_term_cache ) {
+		return;
+	}
+
+	$ids_may_have_uncached_data = array_diff( $ids, $non_cached_ids );
+
+	if ( empty( $ids_may_have_uncached_data ) ) {
+		// No work to do, caches primed within update_post_caches().
+		return;
+	}
+
+	if ( $update_meta_cache ) {
+		$ids_with_uncached_meta = _get_non_cached_ids( $ids_may_have_uncached_data, 'post_meta' );
+		update_postmeta_cache( $ids_with_uncached_meta );
+	}
+
+	if ( $update_term_cache ) {
+		// update_object_term_cache() checks for uncached IDs internally.
+		update_object_term_cache( $ids_may_have_uncached_data, 'post' );
```

Comment:
Hmm, we can't hardcode `post` here as the query can be for "any" post type(s). I guess that's one reason why should consider @spacedmonkey's solution in #3662.

 https://github.com/WordPress/wordpress-develop/pull/3662/files#diff-ef9c5475561d459917dac57d21e08a6dfeb36f8b1c44b978d3696aec8b97c405R7903-R7906